distlib/util.py

```diff
@@ -1435,7 +1435,7 @@ def connect(self):
             context = ssl.SSLContext(ssl.PROTOCOL_SSLv23)
             if hasattr(ssl, 'OP_NO_SSLv2'):
                 context.options |= ssl.OP_NO_SSLv2
-            if self.cert_file:
+            if hasattr(self, 'cert_file') and self.cert_file:
```

Comment:
I think `getattr(self, 'cert_file', None)` would have the same effect, wouldn't it?